### PR TITLE
updating rest client object to v2022_12_01_preview for automl files

### PR DESCRIPTION
**Pull request by guptha23**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/image_vertical/image_object_detection.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 from marshmallow import fields, post_load
 
-from azure.ai.ml._restclient.v2022_10_01_preview.models import (
+from azure.ai.ml._restclient.v2022_12_01_preview.models import (
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/image_vertical/image_object_detection.py:11]: Maybe automl can centralize these imports in some file, so its easier to change next time

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/nlp_vertical/nlp_fixed_parameters.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 from marshmallow import fields, post_load
 
-from azure.ai.ml._restclient.v2022_10_01_preview.models import NlpLearningRateScheduler
+from azure.ai.ml._restclient.v2022_12_01_preview.models import NlpLearningRateScheduler
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/nlp_vertical/nlp_fixed_parameters.py:9]: does sweep job is managed by some other team? I saw some missing updates for sweep

Comment (guptha23) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/nlp_vertical/nlp_fixed_parameters.py:9]: can you please let me know what are the updates that we have missed for sweep? thank you!!

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/image_vertical/image_object_detection.py (hunk 1 above)
Comment (novaturient95) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/image_vertical/image_object_detection.py:11]: This is great suggestion. Will prevent changes across so many files just to use next API version

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/nlp_vertical/nlp_fixed_parameters.py (hunk 2 above)
Comment (MaurisLucis) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/nlp_vertical/nlp_fixed_parameters.py:9]: @wangchao1230 perhaps best for you/Doris as the original modifier to follow up with the other job types for their missed updates? and we can focus on automl in this PR

**Comment by luigiw** [comment]
We have a plan to switch job entity/operations to 2023-02-01-preview to support Singularity. Will 2023-02-01-preview serve the same purpose of this PR for AutoML? If yes, let's join the effort.  @TonyJ1 FYI.

**Comment by TonyJ1** [comment]
> We have a plan to switch job entity/operations to 2023-02-01-preview to support Singularity. Will 2023-02-01-preview serve the same purpose of this PR for AutoML? If yes, let's join the effort. @TonyJ1 FYI.

Thanks @luigiw!
Yes, please let us know if 2023-02-01-preview would work. 

PR: [[ML] Migrate from 2022_12_01_preview API to 2023_02_01_preview API](https://github.com/Azure/azure-sdk-for-python/pull/28863)

Dependency: [MFE Release](https://msdata.visualstudio.com/Vienna/_wiki/wikis/vienna-wiki/39761/AML-API-Release-Schedules) ETA: 02/28

**Comment by guptha23** [comment]
> @guptha23 did you forget the tabular entities folder? ex: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/tabular/automl_tabular.py

sorry, I missed tabular entities folder before. I have updated the automl tabular entities folder now. can you please verify once? thank you!!

**Comment by MaurisLucis** [comment]
I think with this all the automl entities have been covered -- but the CI gates are failing. Final step is ensuring they pass. I took a glance and a few were failing because we import sweep job entities (e.g. early termination policy) and those are still from Oct preview, so it looks like we'll need to upgrade the sweep imports as well at the very least.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/nlp_vertical/nlp_fixed_parameters.py (hunk 2 above)
Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/nlp_vertical/nlp_fixed_parameters.py:9]: Sure. Let's focus on automl in this PR.

**Review by MaurisLucis — APPROVED** [review]

**Comment by harneetvirk** [comment]
Could you please update the changelog for the API version changes?

**Comment by guptha23** [comment]
> Could you please update the changelog for the API version changes?

have updated the changelog with the help of @MaurisLucis. thank you!!

**Review by novaturient95 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/image_vertical/image_object_detection.py (hunk 1 above)
Comment (novaturient95) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/image_vertical/image_object_detection.py:11]: Created a WI: https://msdata.visualstudio.com/Vienna/_workitems/edit/2266774

**Pull request merged**